cli/cmdlineparser.cpp

```diff
@@ -903,8 +903,14 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
 
             // max ctu depth
             else if (std::strncmp(argv[i], "--max-ctu-depth=", 16) == 0) {
-                if (!parseNumberArg(argv[i], 16, mSettings.maxCtuDepth))
+                int temp = 0;
+                if (!parseNumberArg(argv[i], 16, temp))
                     return Result::Fail;
+                if (temp > 10) {
```

Comment:
I would like to keep this cap for a few versions (with a deprecation) because users might have specified a higher value and removing it would lead to different behavior and possibly decreased performance.

That cap previously lived in the function `findPath()` in `lib/ctu.cpp`.